source/internal.d.ts

```diff
@@ -317,3 +320,398 @@ IsPrimitive<Object>
+		? true
+		: boolean;
+
+/**
+Returns a boolean for whether A is false.
+
+@example
+```
+Not<true>
+//=> false
+
+Not<false>
+//=> true
+```
+*/
+export type Not<A extends boolean> = A extends true
+	? false
+	: A extends false
+		? true
+		: boolean;
+
+/**
+Returns the maximum element in an array.
+
+Note:
+- float number is not supported.
+
+@example
+```
+Max<[1, 2, 5, 3]>
+//=> 5
+
+Max<[1, 2, 5, 3, 99, -1]>
+//=> 99
+```
+*/
+export type Max<A extends number[], Result extends number = NegativeInfinity> = A extends [infer F extends number, ...infer R extends number[]]
+	? Gt<F, Result> extends true
+		? Max<R, F>
+		: Max<R, Result>
+	: Result;
+
+/**
+Returns the minimal element in an array.
+
+@example
+```
+Min<[1, 2, 5, 3]>
+//=> 1
+
+Min<[1, 2, 5, 3, -5]>
+//=> -5
+```
+*/
+export type Min<A extends number[], Result extends number = PositiveInfinity> = A extends [infer F extends number, ...infer R extends number[]]
+	? Lt<F & number, Result> extends true
+		? Min<R, F>
+		: Min<R, Result>
+	: Result;
+
+/**
+Calculate the result of `A + B`
+
+Note:
+- A and B can only be small integers.
+- if the result is negative, you can only get `number`.
+
+@example
+```
+Add<111, 222>
+//=> 333
+
+Add<-111, 222>
+//=> 111
+
+Add<111, -222>
+//=> number
+
+Add<PositiveInfinity, -9999>
+//=> PositiveInfinity
+
+Add<PositiveInfinity, NegativeInfinity>
+//=> number
+```
+*/
+// TODO: Support big integer and negative number.
+export type Add<A extends number, B extends number> = [
+	IsEqual<A, PositiveInfinity>, IsEqual<A, NegativeInfinity>,
+	IsEqual<B, PositiveInfinity>, IsEqual<B, NegativeInfinity>,
+] extends infer R extends [boolean, boolean, boolean, boolean]
+	? Or<
+	And<IsEqual<R[0], true>, IsEqual<R[3], false>>,
+	And<IsEqual<R[2], true>, IsEqual<R[1], false>>
+	> extends true
+		? PositiveInfinity
+		: Or<
+		And<IsEqual<R[1], true>, IsEqual<R[2], false>>,
+		And<IsEqual<R[3], true>, IsEqual<R[0], false>>
+		> extends true
+			? NegativeInfinity
+			: true extends R[number]
+				? number
+				: ([IsNegative<A>, IsNegative<B>] extends infer R
+					? [false, false] extends R
+						? [...NewArray<A>, ...NewArray<B>]['length']
+						: [true, true] extends R
+							? number
+							: Max<[Abs<A>, Abs<B>]> extends infer Max_
+								? Min<[Abs<A>, Abs<B>]> extends infer Min_ extends number
+									? Max_ extends A | B
+										? Sub<Max_, Min_>
+										: number
+									: never
+								: never
+					: never) & number
+	: never;
+
+/**
+Calculate the result of `A - B`.
+
+Note:
+- A and B can only be small integers.
+- if the result is negative, you can only get `number`.
+
+@example
+```
+Sub<333, 222>
+//=> 111
+
+Sub<111, -222>
+//=> 333
+
+Sub<-111, 222>
+//=> number
+
+Sub<PositiveInfinity, 9999>
+//=> PositiveInfinity
+
+Sub<PositiveInfinity, PositiveInfinity>
+//=> number
+```
+ */
+// TODO: Support big integer and negative number.
+export type Sub<A extends number, B extends number> = [
```

Comment:
```suggestion
export type Subtract<A extends number, B extends number> = [
```

Reply:
Can you also share your opinion on https://github.com/sindresorhus/type-fest/issues/774 ? (Comment on the issue not here)